test/unit/test_db_os_executor.py

```diff
@@ -0,0 +1,67 @@
+from unittest.mock import MagicMock, Mock
+from exasol_integration_test_docker_environment \
+    .lib.base.db_os_executor import (
+        DockerClientFactory,
+        DbOsExecutor,
+        DockerExecutor,
+        SshExecutor,
+        DockerExecFactory,
+        SshExecFactory,
+)
+from exasol_integration_test_docker_environment.lib.test_environment.ports import Ports
+from exasol_integration_test_docker_environment.lib.data.ssh_info import SshInfo
+from exasol_integration_test_docker_environment.lib.data.database_info import DatabaseInfo
+
+
+def test_docker_client_factory():
+    factory = DockerClientFactory()
+    client = Mock()
+    factory.client = MagicMock(return_value=client)
+    testee = DockerExecFactory("container_name", factory)
+
+    executor = testee.executor()
+    factory.client.assert_called()
+    assert executor._client == client
+
+
+def test_executor_closes_client():
+    container = Mock()
+    containers_mock = Mock()
+    containers_mock.get = MagicMock(return_value=container)
+    client = Mock(containers=containers_mock)
```

Comment:
```suggestion
    client:Union[MagicMock, ContextDockerClient] = create_autospec(ContextDockerClient)
```

Reply:
This does not seem to work.
I need to mock containers_mock.get and also need a dedicated container object in order to verify if method `exec_ _run()` has been called.

Reply:
I think I found a solution: 
```python 
from docker import DockerClient
from docker.models.containers import Container as DockerContainer

def test_executor_closes_client():
    container = create_autospec(DockerContainer)
    client:Union[MagicMock, DockerClient] = create_autospec(DockerClient)
    client.containers.get = MagicMock(return_value=container)
    with DockerExecutor(client, "container_name") as executor:
        executor.exec("sample command")
        container.exec_run.assert_called_with("sample command")
        client.close.assert_not_called()
    client.close.assert_called()
```